ps_facetedsearch.php

```diff
@@ -151,10 +163,14 @@ public function hookProductSearchProvider($params)
         // Query is an instance of:
         // PrestaShop\PrestaShop\Core\Product\Search\ProductSearchQuery
         if ($query->getIdCategory()) {
-            return new Ps_FacetedsearchProductSearchProvider($this);
-        } else {
-            return null;
+            $this->context->controller->addJS($this->_path . 'views/dist/front.js');
```

Comment:
It prevent the usage of adding useless css & js when there's no product to display.

Reply:
but you can check in which page you are in the `header` hook so that it only include assets when needed
Anyway as long as it works it's good, it's just that I believe `addJs/addCss` needs to be called early enough in the workflow in order to work, but if it works with this one great!

Reply:
It needs to be display everytime this hook is called :P